classes/Cart.php

```diff
@@ -4088,8 +4101,7 @@ public function checkQuantities($returnProductOnFailure = false)
              */
             if (
                 !$product['active'] ||
-                !$product['available_for_order'] ||
-                (!$product['allow_oosp'] && $product['stock_quantity'] < $product['cart_quantity'])
```

Comment:
This piece of code is not needed, because it uses only the quantity from stock_available. We need the live one, which is checked few lines below, properly. Because we pass $this (Cart) also, it also checks the amount of products in cart in deep_quantity.